base/algorithms.jl

```diff
@@ -0,0 +1,119 @@
+module Algorithms
```

Comment:
But where to put the general methods which work on `Union{AbstractArray,Associative,AbstractSet}` ?